trail/upload/process_uploads/header_standardizer.py

```diff
@@ -0,0 +1,329 @@
+    * converting WCS information contained in each image-like header such as
+      PrimaryHDU, CompImageHDu etc. (i.e. each header for which it is infered
+      by its Astropy type will contain image under its `data` attribute; as
+      opposed to BinTableHDU etc.) into a standard set of keys as defined by
+      the database models.
+    * converting a set of header keywords into metadata keys as defined by the
+      database models. The mandatory data consists of observatory location,
+      instrument description and time of observation while the remaining
+      metadata may vary.
+
+
+    Parameters
+    ----------
+    header : `object`
+         The header, Astropy HDU and its derivatives.
+    \**kwargs : `dict`
+       Additional keyword arguments
+
+    Keyword arguments
+    -----------------
+    filename : `str`
+        Name of the file from which the HDU was read from, sometimes can encode
+        additional metadata.
+    """
+
+    standardizers = dict()
+    """All registered header standardizers."""
+
+    name = None
+    """Standardizer's name. Only named standardizers will be registered."""
+
+    priority = 0
+    """Priority. Standardizers with high priority are prefered over
+    standardizers with low priority when processing header metadata.
+    """
+
+    @abstractmethod
+    def __init__(self, header, **kwargs):
+        self.header = header
+        self._kwargs = kwargs
+
+    def __init_subclass__(cls, **kwargs):
+        name = getattr(cls, "name", False)
+        if name and name is not None:
+            super().__init_subclass__(**kwargs)
+            HeaderStandardizer.standardizers[cls.name] = cls
+
+    @staticmethod
+    def _computeStandardizedWcs(header, dimX, dimY):
+        """Given an Header containing WCS data and the dimensions of an image
+        calculates the values of world coordinates at image corner and image
+        center and projects them to a unit sphere in Cartesian coordinate
+        system.
+
+        Parameters
+        ----------
+        header : `object`
+            The header, Astropy HDU and its derivatives.
+        dimX : `int`
+            Image dimension in x-axis.
+        dimY : `int`
+            Image dimension in y-axis.
+
+        Returns
+        -------
+        standardizedWcs : `dict`
+            Calculated coorinate values, a dict with wcs_radius,
+            wcs_center_[x, y, z] and wcs_corner_[x, y, z]
+
+        Notes
+        -----
+        The center point is the center of the image as determined its
+        dimensions directly or via header keywords NAXIS1 and NAXIS2, and the
+        corner is taken to be the (0,0)-th pixel.
+        World coords. at the points are to unit sphere, in Cartesian system,
+        and the distance between them is calculated.
+        """
+        standardizedWcs = {}
+        centerX, centerY = int(dimX/2), int(dimY/2)
+
+        # TODO: When eventually logging is added to processing, redirect these
```

Comment:
This one *definitely* needs an issue opened if we don't have one... I would argue we might initially want to give up if we can't make sense of the header WCS. Maybe later we can get smarter. Won't silently ignoring this lead to a metadata table full of useless WCS-less things?

Reply:
See the comment about error handling [above](https://github.com/dirac-institute/trailblazer/pull/10#discussion_r654729848).

In short: absolutely yes, but it was not wholly clear to me what data is "good enough" to commit and what not.